config/config.go

```diff
@@ -0,0 +1,56 @@
+package config
+
+import (
+	"errors"
+	"io/fs"
+)
+
+// CertificateReference is a named file path.
+type CertificateReference struct {
+	Name string `json:"name"`
+	Path string `json:"path"`
+}
+
+// Is checks whether the given name is equal with the Name variable
+func (c CertificateReference) Is(name string) bool {
+	return c.Name == name
+}
+
+// Config reflects the config file.
+// Specification: https://github.com/notaryproject/notation/pull/76
+type Config struct {
+	VerificationCertificates VerificationCertificates `json:"verificationCerts"`
+	InsecureRegistries       []string                 `json:"insecureRegistries"`
+	CredentialsStore         string                   `json:"credsStore,omitempty"`
+	CredentialHelpers        map[string]string        `json:"credHelpers,omitempty"`
+}
+
+// VerificationCertificates is a collection of public certs used for verification.
+type VerificationCertificates struct {
+	Certificates []CertificateReference `json:"certs"`
+}
+
+// NewConfig creates a new config file
+func NewConfig() *Config {
+	return &Config{
+		InsecureRegistries: []string{},
+	}
+}
+
+// Save stores the config to file
+func (c *Config) Save() error {
```

Comment:
Maybe I remebered it wrong. But was there an interface which defines `Save`? Did it get removed? My only concern is that when `notation` uses it, we have to mock the fileSystem to do a real save operation for testing.

Reply:
There are 2 paths defined for mocking.

> var (
> 	// ConfigPath is the path for config.json
> 	ConfigPath string
> 	// SigningKeysPath is the path for signingkeys.json
> 	SigningKeysPath string
> )

You can change those 2 paths to be temp directories for testing `Save()`